### PR TITLE
[improve][doc] Improve the information architecture of the security chapter

### LINKED ISSUE #17561: [Doc] Authentication documentation mismatch / outdated
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### What issue do you find in Pulsar docs?

The documentation regarding authentication methods is mismatched/outdated.

On the security overview page, it lists most providers (it is missing SASL though):
https://pulsar.apache.org/docs/2.10.1/security-overview/#authentication-providers

On the java client page, it lists only three providers:
https://pulsar.apache.org/docs/2.10.1/client-libraries-java/#authentication

### What is your suggestion?

If you go to the security overview page and select an authentication method you will reach a page that also contains a code configuration example:
https://pulsar.apache.org/docs/2.10.1/security-tls-authentication/#java-client

I think it would be good to remove the list on the java client page and just link to the security page. 
Another option would be to update the java client page (but I think it will go out of sync again at some point if we maintain it in two places)

### Any reference?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by momo-jun**
### Motivation

Improve the doc set of the `Security` chapter to provide Pulsar users with a more accurate, easy-to-read, and comprehensive information experience.

Note: This PR is a quick change to the overall IA of security topics. I will include more detailed IA changes and content changes for encryption and authentication in a couple of follow-up PRs, such as https://github.com/apache/pulsar/pull/17666.

### Modifications

1. Group encryption- and authentication-related topics in the sidebar (titles of some topics can be improved later).
    
image ----> image

2. Move `TLS encryption/authentication using Keystore` sections to related parent topics, and remove `Use TLS with KeyStore` topic.
3. Streamline the headings of authentication topics.
4. Use language tabs to group code snippets throughout the security chapter.
5. Remove the java code snippets from the java client page, and add links to individual authentication topics - fix #17561.
6. Add code snippet for python clients to support basic auth #17482.
7. Add code snippet for go clients to support basic auth https://github.com/apache/pulsar-client-go/pull/778.

Preview looks good. I've checked all the topics and here is one of the screenshots.
image

### Documentation
- [x] `doc`

### CODE REVIEW

File: site2/docs/client-libraries-java.md [hunk 1]
```diff
@@ -1258,88 +1258,14 @@ For examples of ProtobufNativeSchema, see [`SchemaDefinition` in `Complex type`]
 
 ## Authentication
 
-Pulsar currently supports three authentication schemes: [TLS](security-tls-authentication.md), [Athenz](security-athenz.md), and [Oauth2](security-oauth2.md). You can use the Pulsar Java client with all of them.
+Pulsar currently supports three authentication providers:
+* [TLS](security-tls-authentication.md#configure-tls-authentication-in-pulsar-clients)
+* [JWT](security-jwt.md#configure-jwt-authentication-in-pulsar-clients)
+* [Athenz](security-athenz.md#configure-athenz-authentication-in-pulsar-clients)
+* [Kerberos](security-kerberos.md#java-client-and-java-admin-client)
+* [Oauth2](security-oauth2.md#configure-oauth2-authentication-in-pulsar-clients)
```
Comment (nodece) [inline site2/docs/client-libraries-java.md:1266]:
```suggestion
* [OAuth2](security-oauth2.md#configure-oauth2-authentication-in-pulsar-clients)
```

File: site2/docs/security-jwt.md [hunk 2]
```diff
@@ -282,3 +163,122 @@ brokerClientAuthenticationParameters={"token":"eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ0
...
+
+client = Client('pulsar://broker.example.com:6650/'
+                authentication=AuthenticationToken(read_token))
+```
+
+</TabItem>
+<TabItem value="Go">
+
+```go
+client, err := NewClient(ClientOptions{
```
Comment (nodece) [inline site2/docs/security-jwt.md:242]:
```suggestion
client, err := pulsar.NewClient(pulsar.ClientOptions{
```

File: site2/docs/security-jwt.md [hunk 3]
```diff
@@ -282,3 +163,122 @@ brokerClientAuthenticationParameters={"token":"eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ0
...
+	URL:            "pulsar://localhost:6650",
+	Authentication: NewAuthenticationToken("eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJKb2UifQ.ipevRNuRP6HflG8cFKnmUPtypruRC4fb1DWtoLL62SY"),
+})
+```
+
+Similarly, you can also pass a `Supplier`:
+
+```go
+client, err := NewClient(ClientOptions{
```
Comment (nodece) [inline site2/docs/security-jwt.md:251]:
```suggestion
client, err := pulsar.NewClient(pulsar.ClientOptions{
```

File: site2/docs/security-jwt.md [hunk 4]
```diff
@@ -282,3 +163,122 @@ brokerClientAuthenticationParameters={"token":"eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ0
...
+	URL:            "pulsar://localhost:6650",
+	Authentication: NewAuthenticationTokenSupplier(func () string {
```
Comment (nodece) [inline site2/docs/security-jwt.md:253]:
```suggestion
	Authentication: pulsar.NewAuthenticationTokenSupplier(func () string {
```

File: site2/docs/security-tls-transport.md [hunk 5]
```diff
@@ -324,3 +322,258 @@ var client = PulsarClient.Builder()
 ```
 
 > Note that `VerifyCertificateName` refers to the configuration of hostname verification in the C# client.
+
+</TabItem>
+</Tabs>
+````
+
+## Configure TLS encryption using CLI tools
```
Comment (nodece) [inline site2/docs/security-tls-transport.md:330]:
```suggestion
## Configure TLS encryption in CLI tools
```

**Review by nodece — COMMENTED** [review]
Overall looks good to me.

**Comment by momo-jun** [comment]
@nodece thanks for the review. Your comments have been addressed.

**Review by nodece — APPROVED** [review]
LGTM

**Comment by momo-jun** [comment]
@daveduggins This is the improvement I mentioned on this Wednesday. Feel free to take a look and comment.

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**